### DISCUSSION TITLE
Couldn't create new column with upper case

### DISCUSSION THREAD

**Discussion by vlacko0930** [dbeaver/dbeaver#11969] [Q&A]
#### System information: 
- Operating system: Windows 10
- DBeaver version: 7.3.1

#### Connection specification:
- Database name and version: PostgreSQL 12.5
- PostgreSQL JDBC Driver 42.2.5

#### Describe the problem you're observing:
When I'd like to create a new column in an existing table with GUI, it always changes the upper cases in name to lower cases after I close the modalwindow.

**Comment by ShadelessFox** [comment]
Hello @vlacko0930,

You can put the name into double quotes like so: <kbd>"NaMe"</kbd>.

Quote from [PostgreSQL documentation](https://www.postgresql.org/docs/current/sql-syntax-lexical.html):
> Quoting an identifier also makes it case-sensitive, whereas **unquoted names are always folded to lower case**. For example, the identifiers FOO, foo, and "foo" are considered the same by PostgreSQL, but "Foo" and "FOO" are different from these three and each other. (The folding of unquoted names to lower case in PostgreSQL is incompatible with the SQL standard, which says that unquoted names should be folded to upper case. Thus, foo should be equivalent to "FOO" not "foo" according to the standard. If you want to write portable applications you are advised to **always quote a particular name or never quote it**.)

We do that for various renaming actions, but the fact that we don't do that for object creation is silly.

**Comment by serge-rider** [comment]
Yep.
We perform explicit case conversion to the database server default case by default. 
You need to enclose in quotes to tell dbeaver that you mean it. 

Otherwise too many people create case-sensitive identifiers by mistake.

**Reply by vlacko0930** [reply]
Ok, but the GUI will be better, if it could be set in window preferences. :-)

**Reply by ShadelessFox** [reply]
You can, actually. Please refer to the <kbd>Preferences</kbd> &rArr; <kbd>Connections</kbd> &rArr; <kbd>Metadata</kbd> &rArr; <kbd>Use case-sensitive names in DDL statements</kbd>.

Also, you can specify this option per connection rather than changing it globally :)
